shared-actions/npm-setup/action.yml

```diff
@@ -0,0 +1,31 @@
+name: JS/TS environment setup
```

Comment:
Opened to suggestion for name (the folder is currently called npm-setup, but I feel it does more than that)

Reply:
I like this name 👍

Reply:
cool, I'll change the folder name for `js-ts-environment-setup` then